### PR TITLE
Fixes #4176 electrode diffusivity rename

### LINKED ISSUE #4176: [Bug]: "electrode diffusivity" -> "particle diffusivity" is still breaking change on `develop` when using pre-24.5 pybamm.ParameterValues
### PyBaMM Version

develop

### Python Version

3.9.13

### Describe the bug

Since #4072 re-inverted the logic fix in #3871, the release candidate on `develop` causes the rename of "electrode diffusivity" -> "particle diffusivity" to be a breaking change. Parameter sets using the old key names do not simulate in the 24.5 RC.

### Steps to Reproduce

```python
import pybamm

parameter_values = pybamm.ParameterValues("Chen2020")

# Dummy a pybamm.ParameterValues object with a legacy parameter name
parameter_values.update(
    {
        "Negative electrode diffusivity [m2.s-1]": parameter_values["Negative particle diffusivity [m2.s-1]"],
    },
    check_already_exists=False,
)
del parameter_values["Negative particle diffusivity [m2.s-1]"]

# Try to run a simulation with the old-style parameter set
model = pybamm.lithium_ion.DFN()
experiment = pybamm.Experiment(
    [
        "Discharge at 1C until 3 V",
    ]
)
sim = pybamm.Simulation(
    model,
    parameter_values=parameter_values,
    experiment=experiment,
)

# KeyError provoked here on call into pybamm.ParameterValues with new key name
sol = sim.solve()
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by BradyPlanden**
# Description

Fixes #4176. In addition to this issue, in the `develop` implementation changes to `electrode diffusivity` via `parameter_values.update()` will not be captured. To account for this, the I've taken the depreciated key and appended the new key into parameter_set dictionary in the event that the depreciated parameter is supplied. This also cleans up the `bpx.py` implementation since the logic for changing the key is moved into the parameters class.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [X] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
Hi @ejfdickinson, would be able to try these changes out? Also, thanks for flagging this at a reasonable time before the release and thank you @BradyPlanden for taking it!

File: tests/unit/test_parameters/test_bpx.py [hunk 1]
```diff
@@ -107,28 +109,51 @@ def setUp(self):
...
+        )
 
-            pv = pybamm.ParameterValues.create_from_bpx(tmp.name)
+        filename = "tmp.json"
+        sols = []
+        for obj in bpx_objs:
+            with tempfile.NamedTemporaryFile(
+                suffix=filename, delete=False, mode="w"
+            ) as tmp:
+                # write to a tempory file so we can
```
Comment (agriyakhetarpal) [inline tests/unit/test_parameters/test_bpx.py:141]:
```suggestion
                # write to a temporary file so we can
```

**Review by agriyakhetarpal — COMMENTED** [review]
Just one nitpicky typo suggestion – I don't understand the BPX code enough so I will leave it to someone else to approve but the rest of the changes look good to me, thanks!

**Comment by ejfdickinson** [comment]
@agriyakhetarpal @BradyPlanden The test I made is passing with these changes, so seems ok to me. Note also typoes in the changelog!

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 ## Bug Fixes
 
+- Fixed bug where passing depreciated `electrode diffusivity` parameter resulted in a breaking change and/or the corresponding diffusivity parameter not updating. Improved th the depreciated translation around BPX.
```
Comment (brosaplanella) [inline CHANGELOG.md:27]:
```suggestion
- Fixed bug where passing deprecated `electrode diffusivity` parameter resulted in a breaking change and/or the corresponding diffusivity parameter not updating. Improved the deprecated translation around BPX. ([#4176](https://github.com/pybamm-team/PyBaMM/pull/4176))
```

**Review by brosaplanella — APPROVED** [review]
Looks good to me, especially if it fixes @ejfdickinson example.

**Pull request merged**